InteractionBase/src/commands/SwitchViewCommand.cpp

```diff
@@ -0,0 +1,62 @@
+/***********************************************************************************************************************
+ **
+ ** Copyright (c) 2011, 2014 ETH Zurich
+ ** All rights reserved.
+ **
+ ** Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+ ** following conditions are met:
+ **
+ **    * Redistributions of source code must retain the above copyright notice, this list of conditions and the
+ **      following disclaimer.
+ **    * Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+ **      following disclaimer in the documentation and/or other materials provided with the distribution.
+ **    * Neither the name of the ETH Zurich nor the names of its contributors may be used to endorse or promote products
+ **      derived from this software without specific prior written permission.
+ **
+ **
+ ** THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ ** INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ ** DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ ** SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ ** SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ ** WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ ** OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ **
+ **********************************************************************************************************************/
+
+#include "SwitchViewCommand.h"
+#include "VisualizationBase/src/items/ViewItem.h"
+
+namespace Interaction {
+
+SwitchViewCommand::SwitchViewCommand()
+	:MenuCommand("switch", QStringList({"name"}))
+{
+}
+
+bool SwitchViewCommand::canUseTarget(Visualization::Item *, Visualization::Item *,
+		const std::unique_ptr<Visualization::Cursor>&)
+{
+	return true;
+}
+
+CommandResult* SwitchViewCommand::executeWithArguments(Visualization::Item *, Visualization::Item *target,
+		const QStringList& arguments, const std::unique_ptr<Visualization::Cursor>&)
+{
+	QString name = arguments.at(0);
+	auto view = target->scene()->getViewItem(name);
+	if (view)
+	{
+		target->scene()->switchToView(view);
```

Comment:
I am in favor of having an overload of Scene::switchToView, which directly takes a QString viewName and tries to switch if such a view exists. This will make the action easily accessible also from other code.